meilisearch-schema/src/schema.rs

```diff
@@ -260,7 +263,13 @@ impl Schema {
             self.indexed = match self.indexed.take() {
                 // valid because indexed is All and indexed() return the content of
                 // indexed_map that is already updated
-                OptionAll::All => OptionAll::Some(self.indexed().into_owned()),
+                OptionAll::All => {
+                    let indexed = self.indexed_map
+                        .iter()
+                        .filter_map(|(&i, _)| if i != id { Some(i) } else { None })
+                        .collect();
+                    OptionAll::Some(indexed)
```

Comment:
Are you just search for `id` in this `HashSet` ? Why not `get`ing it?

Reply:
Whoops, no, you are search for all others, sorry, closing!

Reply:
no, I am collecting all the fields in the indexed_map (`HashMap`) except the `id` one, and collect all this in a `HashSet`. I could collect all in an `HashSet` too and `HashSet::remove(id)` too:

```rust
                    let mut indexed: HashSet<_> = self.indexed_map
                        .keys()
                        .cloned()
                        .collect();
                    indexed.remove(&id);
                    OptionAll::Some(indexed)
```
do you prefer that?

Reply:
I would maybe prefer you just `filter` on `keys` instead of a `filter_map`.

```rust
self.indexed_map
                        .keys()
                         .filter(|i| i != id)
                        .cloned()
                        .collect();
```

Reply:
well well, turns out it isn't necessary:
```rust
            self.indexed_map.remove(&id);
```
is done just before